File: examples/metal_transfert/drop_python.py
import sys
sys.path.append('../../libs')
from PyerCv import *
from numpy import arange

an=erMetalTransfertAnalysis("peo","peo")
#zone d etudes
rect=CvRect();
rect.x =100;rect.y=120;rect.width=70;rect.height=70;

# Premier jeu de parametres
p1  = erSmootP(BLUR,3)
p2  = erSmootP(GAUSSIAN,9);
cann= erCannyP(355,355);
adp = erAdThrP(THRESH_BINARY,AM_MEAN,51,11,255);
alp = erAlphaP(1.)

# Immediatrement apres le detachement on a un jeu de parametres
an.defineParameters(rect,p1,p2,cann,adp,alp)
an.doIt("imageExemple_1.bmp")

# On change pour les images en vol
p1  = erSmootP(BLUR,7)
p2  = erSmootP(GAUSSIAN,7);
adp = erAdThrP(THRESH_BINARY,AM_MEAN,51,17,255);
alp = erAlphaP(2.)
an.defineParameters(rect,p1,p2,cann,adp,alp)
for i in arange(2,7,1):
    an.doIt("imageExemple_"+str(i)+".bmp")

File: examples/metal_transfert/drop_python.cpp
#include<erCv/erCvAnalysis.hpp>


/* Valeurs des paramettres a introduire, pour chaque fonction:
Smooth           : 7, 1 (0 pour OK)
AdaptiveThreshold: 1, 29, 1, 1 (0 pour OK)
Smooth2          : 7, 3 (0 pour OK)
Canny            : 355, 355 



/* Paramettres a introduir par l'usager dans l'appel du programe: */
/* 1- NOM choisi pour identifier les ficher de sortie,*/
/* 2- Adresse et nom de la premiere image a traiter*/





int main( int hola, char** file_name)
{ 
  CvRect rect;
  //rect.x=100;rect.y=120;rect.width=70;rect.height=70;
  rect.x=0;rect.y=0;rect.width=255;rect.height=255;
  erSmootP p1(BLUR,5),p2(GAUSSIAN,5);
  erCannyP cann(355,355);
  erAdThrP adp(THRESH_BINARY,AM_MEAN,53,19,255); //** <
  erAlphaP alp(1.);
  erMetalTransfertAnalysis emt("peo","peo");
  emt.defineParameters(rect,p1,p2,cann,adp,alp);
  emt.doIt("imageExemple_1.bmp");
  emt.doIt("imageExemple_2.bmp");
  emt.doIt("imageExemple_3.bmp");
  emt.doIt("imageExemple_4.bmp");
  emt.doIt("imageExemple_5.bmp");
  emt.doIt("imageExemple_6.bmp");
  return(0);
}
